packages/middleware-token/src/resolveTokenConfig.ts

```diff
@@ -5,7 +5,9 @@ import { tokenDefaultProvider } from "./tokenDefaultProvider";
 /**
  * @internal
  */
-export const resolveTokenConfig = <T>(input: T & TokenInputConfig): T & TokenResolvedConfig => ({
-  ...input,
-  token: input.token ? normalizeTokenProvider(input.token) : tokenDefaultProvider(input as any),
-});
+export const resolveTokenConfig = <T>(input: T & TokenInputConfig): T & TokenResolvedConfig => {
+  const { token } = input;
+  return Object.assign(input, {
+    token: token ? normalizeTokenProvider(token) : tokenDefaultProvider(input as any),
```

Comment:
this fallback makes no sense to me since `input` has no overlap with `FromSsoInit`, the type of the falsy branch.